hi-ml-multimodal/notebooks/phrase_grounding.ipynb

```diff
@@ -59,64 +98,91 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "# Load the text inference engine\n",
     "text_inference = get_cxr_bert_inference()\n",
-    "\n",
-    "# Load the image inference engine\n",
-    "resnet_checkpoint_path = \"\"  # add path to checkpoint here\n",
-    "if not Path(resnet_checkpoint_path).is_file():\n",
-    "    print(\"Checkpoint file not found!\")\n",
-    "    resnet_checkpoint_path = None\n",
-    "image_inference = ImageInferenceEngine(\n",
-    "    image_model=ImageModel(img_model_type=\"resnet50\", joint_feature_size=128, pretrained_model_path=resnet_checkpoint_path),\n",
-    "    transform=create_chest_xray_transform_for_inference(resize=512, center_crop_size=480))\n",
-    "\n",
-    "# Instantiate the joint inference engine\n",
+    "image_inference = get_cxr_resnet_inference()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Instantiate the joint inference engine:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
     "image_text_inference = ImageTextInferenceEngine(\n",
     "    image_inference_engine=image_inference,\n",
     "    text_inference_engine=text_inference,\n",
     ")"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Helper visualization functions"
+   ]
+  },
   {
    "cell_type": "code",
    "execution_count": null,
    "metadata": {},
    "outputs": [],
    "source": [
-    "\n",
     "def plot_phrase_grounding(image_path: Path, text_prompt: str) -> None:\n",
-    "    sim_map = image_text_inference.get_similarity_map_from_raw_data(image_path=image_path, query_text=text_prompt)\n",
-    "    plot_phrase_grounding_similarity_map(image_path=image_path, similarity_map=sim_map)\n",
+    "    similarity_map = image_text_inference.get_similarity_map_from_raw_data(\n",
+    "        image_path=image_path,\n",
+    "        query_text=text_prompt,\n",
+    "        interpolation=\"bilinear\",\n",
+    "    )\n",
+    "    plot_phrase_grounding_similarity_map(\n",
+    "        image_path=image_path,\n",
+    "        similarity_map=similarity_map,\n",
+    "    )\n",
     "\n",
     "def plot_phrase_grounding_from_url(image_url: str, text_prompt: str) -> None:\n",
-    "    image_path = Path(tempfile.tempdir, 'downloaded_chest_xray.jpg')\n",
+    "    image_path = Path(tempfile.tempdir, \"downloaded_chest_xray.jpg\")\n",
     "    !curl -s -L -o {image_path} {image_url}\n",
     "    plot_phrase_grounding(image_path, text_prompt)"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Inference\n",
+    "\n",
+    "We will run inference on a chest X-ray from [Radiopaedia](https://radiopaedia.org/), but any can be used."
```

Comment:
but any other chest-xray image in DICOM or JPEG format can be used for research purposes.